src/main/kotlin/org/cascadebot/cascadebot/utils/SerializableMessage.kt

```diff
@@ -0,0 +1,93 @@
+        var mentionsEveryone: Boolean,
+        var userMentions: List<Long>,
+        var channelMentions: List<ChannelMention>,
+        var roleMentions: List<Long>,
+        var attachments: List<Attachment>,
+        var embeds: List<WebhookEmbed>,
+        var reactions: List<Reaction>,
+        var pinned: Boolean,
+        var messageType: MessageType,
+        var messageFlags: List<MessageFlag>?
+) {
+    companion object {
+        @JvmStatic
+        fun createSerializeMessageFromJda(message: Message): SerializableMessage? {
+            val userMentions: List<Long> = message.mentionedUsers.stream().map { it.idLong }.collect(Collectors.toList())
+            val channelMentions: List<ChannelMention> = message.mentionedChannels.stream().map { ChannelMention(it.idLong, it.guild.idLong) }.collect(Collectors.toList())
+            val roleMentions: List<Long> = message.mentionedRoles.stream().map { it.idLong }.collect(Collectors.toList())
+            val attachments: List<Attachment> = message.attachments.stream().map { Attachment(it.idLong, it.fileName, it.size, it.url) }.collect(Collectors.toList())
+            val embeds: List<WebhookEmbed> = message.embeds.stream().map {
+                var webhookEmbedBuilder = WebhookEmbedBuilder()
+                if (it.title != null) {
+                    webhookEmbedBuilder.setTitle(WebhookEmbed.EmbedTitle(it.title!!, it.url))
+                }
+                if (it.description != null) {
+                    webhookEmbedBuilder.setDescription(it.description)
+                }
+                if (it.timestamp != null) {
+                    webhookEmbedBuilder.setTimestamp(it.timestamp)
+                }
+                webhookEmbedBuilder.setColor(it.colorRaw)
+                if (it.footer != null) {
+                    webhookEmbedBuilder.setFooter(WebhookEmbed.EmbedFooter(it.footer!!.text!!, it.footer!!.iconUrl))
+                }
+                if (it.image != null) {
+                    webhookEmbedBuilder.setImageUrl(it.image!!.url)
+                }
+                if (it.thumbnail != null) {
+                    webhookEmbedBuilder.setThumbnailUrl(it.thumbnail!!.url)
+                }
+                if (it.author != null) {
+                    webhookEmbedBuilder.setAuthor(WebhookEmbed.EmbedAuthor(it.author!!.name!!, it.author!!.iconUrl, it.author!!.url))
+                }
+                for (field: MessageEmbed.Field in it.fields) {
+                    webhookEmbedBuilder.addField(WebhookEmbed.EmbedField(field.isInline, field.name!!, field.value!!))
+                }
+                webhookEmbedBuilder.build()
+            }.collect(Collectors.toList())
+            val reactions: List<Reaction> = message.reactions.stream().map { Reaction(it.count, it.reactionEmote.idLong, it.reactionEmote.name) }.collect(Collectors.toList())
+            return SerializableMessage(message.idLong, message.channel.idLong, message.guild.idLong, message.author.idLong,
```

Comment:
that's one long constructor 😂

Reply:
I don't really have a better option lol.